base/irrationals.jl

```diff
@@ -76,12 +77,23 @@ end
 <(x::Float32, y::AbstractIrrational) = x <= Float32(y,RoundDown)
 <(x::AbstractIrrational, y::Float16) = Float32(x,RoundUp) <= y
 <(x::Float16, y::AbstractIrrational) = x <= Float32(y,RoundDown)
-<(x::AbstractIrrational, y::BigFloat) = setprecision(precision(y)+32) do
-    big(x) < y
-end
-<(x::BigFloat, y::AbstractIrrational) = setprecision(precision(x)+32) do
-    x < big(y)
+
+function cmp(x::BigFloat, y::AbstractIrrational)
+    if cmp(x, Float64(y, RoundDown)) < 0
+        return -1
+    elseif cmp(x, Float64(y, RoundUp)) > 0
+        return +1
+    else
+        for prec in Iterators.countfrom(precision(x), 32)
```

Comment:
This seems dangerous, since if someone defines an "irrational" constant that is actually rational it will never terminate.

How about just something `for prec in (precision(x), max(precision(x), precision(BigFloat)) + 32)` and return `0` if both of these report equality.

Reply:
I'm not sure I agree: that would only occur if (a) you define an irrational as a rational value, and (b) compare it to that same rational. I'm not really sure.

Reply:
See #26550 for why some `AbstractIrrational` subtypes may define rational values.